### ISSUE TITLE
New file share added to function when using azurerm 2.77.0 or greater

### ISSUE THREAD

**Issue by ronorato** [hashicorp/terraform-provider-azurerm#13536]
### Terraform (and AzureRM Provider) Version
Terraform version 1.0.7
Azurerm version 2.77.0

### Affected Resource(s)

azurerm_function_app

* `azurerm_XXXXX`

### Terraform Configuration Files

```
terraform {
  required_providers {
    azurerm = {
      source = "hashicorp/azurerm"
      version = "= 2.77.0"
    }
  }
}

provider azurerm {
    features {}
}

### Resource Groups
resource azurerm_resource_group rg {
  name     = "azurerm-issue-rg"
  location = "centralus"
}

### Service plans
resource azurerm_app_service_plan asp {
    name                = "ep1-asp-01"
    location            = azurerm_resource_group.rg.location
    resource_group_name = azurerm_resource_group.rg.name
    kind                = "elastic"

    sku {
      tier     = "ElasticPremium"
      size     = "EP1"
      capacity = 1
    }
}

resource azurerm_storage_account storage_account {
  name                     = "azurermfuncapp"
  location                 = azurerm_resource_group.rg.location
  resource_group_name      = azurerm_resource_group.rg.name
  account_tier             = "Standard"
  account_replication_type = "LRS"
  tags = {
    "type" = "function"
  }
}

resource azurerm_function_app app {
  name                       = "azurerm-func-app-9282021"
  location                   = azurerm_resource_group.rg.location
  resource_group_name        = azurerm_resource_group.rg.name
  app_service_plan_id        = azurerm_app_service_plan.asp.id
  storage_account_name       = azurerm_storage_account.storage_account.name
  storage_account_access_key = azurerm_storage_account.storage_account.primary_access_key
  version                    = "~3"

  identity {
    type = "SystemAssigned"
  }
}
```

### Debug Output

The above configuration, along with version 2.77.00 will add the following two file shares to the Azure Function Storage.
azurerm-func-app-92820215d13
azurerm-func-app-9282021ff36

I would have expected to see only one file share with the name (which is what azurerm 2.76.00 produces)
azurerm-func-app-9282021-content

I believe that this issue may have been introduced with the change for https://github.com/hashicorp/terraform-provider-azurerm/pull/13349

**Comment by BadgerCode** [comment]
> On top of that, any future change to the function app, even a value in the appSettings will create a new empty file share folder and update the function's WEBSITE_CONTENTSHARE to point to it.

This is quite a critical issue.

Adding/removing/changing an app setting on my function app using AzureRM 2.78.0 effectively kills the function app.
All of the endpoints become unavailable (404) since the host can no longer find the code I deployed to the function.

Reverting back to AzureRM 2.75.0 fixes the issue (potentially 2.76.0 is fine too).

**Comment by ronorato** [comment]
Agreed.  This is a breaking issue for Azure Function Apps, as it impacts both the creation and modification.

We found the issue while troubleshooting a deployment issue in a lower environment.  The team had made a simple configuration change to a couple of the Azure Functions and ran them through the CI/CD pipelines.  The Terraform code was set up to pull in the latest Azure provider, this has since been changed, which cause the pipelines to use AzureRM2.78.0.

After a bit of debugging, we identified that the bug was introduced in 2.77.0.  I believe that the bug may have been introduced with the fix for issue #13580.  I am not sure how any of this got past some basic Function App testing, since the issue appears when either a new function app is created or an existing one is modified.
